devito/types/dimension.py

```diff
@@ -305,6 +305,10 @@ def __new__(cls, *args, **kwargs):
     def __init_finalize__(self, name, spacing=None):
         self._spacing = spacing or Scalar(name='h_%s' % name, is_const=True)
 
+    @cached_property
+    def _indexify_map(self):
```

Comment:
Initially, the design of this and the property appearing in `DefaultDimension` seem a bit weird to me. Will take a closer look though.

Reply:
This property basically say "what should I do when I'm used in an index" which I think beongs here not in a dict built in indexify.